examples/src/bin/ktls-client.rs

```diff
@@ -0,0 +1,666 @@
+//! This is an example showing how to use rustls' external connection API in
```

Comment:
I'm not sure we want to maintain all this stuff in this crate...

Reply:
That's reasonable. This is really the example I wished I had when I was trying to figure out kTLS. I'm fine if it lives elsewhere as long as I can link it in the docs.

If you would rather not maintain it then let me know and I'll move it to its own repo.

Reply:
I think we should use this example as inspiration when integrating this new API into the ktls crate, but probably not keeping it in this repo.

Reply:
I have removed the example. Depending on how things go I may follow up later with a link to either a separate example repo or the ktls crate in a follow-up PR.